semgrep/semgrep/rule_lang.py

```diff
@@ -328,11 +334,15 @@ def construct_object(self, node: Node, deep: bool = False) -> YamlTree:
     yaml = YAML()
     yaml.Constructor = SpanPreservingRuamelConstructor
     data = yaml.load(StringIO(contents))
+
+    schema_path = Path(__file__).parent / "rule_schema.yaml"
+    with schema_path.open() as fd:
+        validation_schema = yaml.load(fd).unroll()
```

Comment:
Can we store this as a const Python `dict` somewhere? That way we won't have to read something from disk every time we perform validation.

Reply:
I much prefer using the external file, as we can publish it. I'm happy to load it once, though.